testing/nginx-naxsi/nginx.initd

```diff
@@ -23,45 +25,24 @@ start_pre() {
 
 reload() {
 	ebegin "Reloading ${SVCNAME} configuration"
-	start_pre && start-stop-daemon --signal HUP --pidfile $pidfile
+	start_pre && $command -s reload
 	eend $?
 }
 
 reopen() {
 	ebegin "Reopening ${SVCNAME} log files"
-	start-stop-daemon --signal USR1 --pidfile $pidfile
+	$command -s reopen
 	eend $?
 }
 
 upgrade() {
-	start_pre || return 1
-
-	ebegin "Upgrading ${SVCNAME} binary"
-
-	einfo "Sending USR2 to old binary"
-	start-stop-daemon --signal USR2 --pidfile $pidfile
-
-	einfo "Sleeping 3 seconds before pid-files checking"
-	sleep 3
-
-	if [ ! -f $pidfile.oldbin ]; then
-		eerror "File with old pid ($pidfile.oldbin) not found"
-		return 1
-	fi
-
-	if [ ! -f $pidfile ]; then
-		eerror "New binary failed to start"
-		return 1
-	fi
-
-	einfo "Sleeping 3 seconds before WINCH"
-	sleep 3 ; start-stop-daemon --signal 28 --pidfile $pidfile.oldbin
-
-	einfo "Sending QUIT to old binary"
-	start-stop-daemon --signal QUIT --pidfile $pidfile.oldbin
-
-	einfo "Upgrade completed"
+	restart
+}
 
-	eend $? "Upgrade failed"
+restart() {
+	rc-service $RC_SVCNAME stop
+	# prevents bind() failed (98: Address in use) error msg
+	sleep 0.05
+	rc-service $RC_SVCNAME start
```

Comment:
Can this be replaced just with `stop && sleep 0.05 && start`? I’m not sure if it can be used with supervisor, I haven’t tried it myself yet.